src/shmem_synchronization.h

```diff
@@ -174,7 +174,7 @@ shmem_internal_fence(shmem_ctx_t ctx)
         }                                                               \
     } while(0)
 
-#if defined(ENABLE_HARD_POLLING)
+#if defined(ENABLE_HARD_POLLING) || defined(ENABLE_MR_ENDPOINT)
```

Comment:
Yes, this is confusing. I think, `FI_PROGRESS_MANUAL` requires polling and we enable manual progress through `FI_MR_ENDPOINT`. We should use a different flag, I guess for manual progress.

Reply:
Ah, I understand now.

Yes, I do think a comment would go a long way here, thanks!  A reference to  the connection between `FI_MR_ENDPOINT` and `FI_PROGRESS_MANUAL` in the ofi transport layer should be sufficient.

Reply:
@davidozog - based on the email conversation with Alexia, I think, it is better not to use the `ENABLE_MR_ENDPOINT` as flag here, which will be confusing as to how MR_ENDPOINT can contribute to progress. I think one way we can handle this is not have a flag for enabling `FI_PROGRESS_MANUAL`. The problem is that we already have `--enable-manual-progress` which does not really enable `FI_PROGRESS_MANUAL` :(. I am open to thoughts / suggestions here.

Reply:
I can think of a few viable options:

1. Rename `--enable-manual-progress` to something like `--enable-progress-probe` and use `--enable-manual-progress` for `FI_MANUAL_PROGRESS`.  I don't like this so much because it's changing the meaning of the flag without user's knowledge.
2. Create a new flag called `--enable-fi-manual-progress`.
4. Change the configury to support something like `ENABLE_WAIT_POLLING` which is enabled for either hard-polling or `MR_ENDPOINT`.
5. Just keep this code as is for now (w/ comments) and fix things in PR #1077

Reply:
I think I like option 2. I was also thinking about that. 

Option 3 is also good. The only thing I am thinking is it does not tell that the `FI_MANUAL_PROGRESS` is being enabled here. As of now, the current SOS supports `FI_MANUAL_PROGRESS` only for `MR_ENDPOINT` and that is only used for this provider. In the future, we might need to revisit if `--enable-fi-manual-progress` is needed but polling based wait is not for any particular provider.

Not trying to convince you, but trying to come up with different reasons.

Reply:
Ah, I do see that issue with option 3, so yes I'm good with option 2.  That was indeed easy to convince me ;).